docs/proposals/super-simple-discovery-api.md

```diff
@@ -0,0 +1,206 @@
+
+This presents a set of problems:
+* It should be super easy for users to configure client systems with a minimum of effort `kubectl` or `kubeadm init` (or other client systems).
+  * Establishing this should be doable even in the face of nodes and master components booting out of order.
+  * We should have mechanisms that don't require users to ever have to manually manage certificate files.
+* Over the life of the cluster this information could change and client systems should be able to adapt.
+
+While this design is mainly being created to help `kubeadm` possible, these problems aren't isolated there and can be used outside of the kubeadm context.
+
+Mature organizations should be able to distribute and manage root certificates out of band of Kubernetes installations.  In that case, clients will defer to corporation wide system installed root certificates or root certificates distributed through other means.  However, for smaller and more casual users distributing or obtaining certificates represents a challenge.
+
+Similarly, mature organizations will be able to rely on a centrally managed DNS system to distribute the location of a set of API servers and keep those names up to date over time.  Those DNS servers will be managed for high availability.
+
+With that in mind, the proposals here will devolve into simply using DNS names that are validated with system installed root certificates.
+
+## Cluster Location information
+
+First we define a set of information that identifies a cluster and how to talk to it.
+
+While we could define a new format for communicating the set of information needed here, we'll start by using the standard [`kubeconfig`](http://kubernetes.io/docs/user-guide/kubeconfig-file/) file format.
+
+It is expected that the `kubeconfig` file will have a single unnamed `Cluster` entry.  Other information (especially authentication secrets) must be omitted.
+
+### Evolving kubeconfig
+
+In the future we look forward to enhancing `kubeconfig` to address some issues.  These are out of scope for this design.  Some of this is covered in [#30395](https://github.com/kubernetes/kubernetes/issues/30395).
+
+Additions include:
+
+* A cluster serial number/identifier.
+  * In an HA world, API servers may come and go and it is necessary to make sure we are talking to the same cluster as we thought we were talking to.
+* A _set_ of addresses for finding the cluster.
+  * It is implied that all of these are equivalent and that a client can try multiple until an appropriate target is found.
+  * Initially I’m proposing a flat set here.  In the future we can introduce more structure that hints to the user which addresses to try first.
+* A _set_ of root certificates to trust for this cluster.
+  * We make this a set so that root certificates can be rotated.
+* Other root certificate options
+  * Trust the system managed list of CAs.
```

Comment:
should already be supported (omit cafile/cadata and leave insecure set to false), though not in addition to a custom ca bundle